src/app/pages/onboarding/set-password/components/password-field.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { FiEye, FiEyeOff } from 'react-icons/fi';
+import { useField } from 'formik';
+import { Box, Button, color, Flex, Input } from '@stacks/ui';
+
+import { ValidatedPassword } from '@app/common/validation/validate-password';
+import { Caption } from '@app/components/typography';
+import { OnboardingSelectors } from '@tests/integration/onboarding/onboarding.selectors';
+
+import { PasswordStrengthIndicator } from './password-strength-indicator';
+import { getIndicatorsOfPasswordStrength } from './password-field.utils';
+
+interface PasswordFieldProps {
+  strengthResult: ValidatedPassword;
+}
+export function PasswordField({ strengthResult }: PasswordFieldProps) {
+  const [field] = useField('password');
+  const [showPassword, setShowPassword] = useState(false);
+
+  const { strengthColor, strengthText } = getIndicatorsOfPasswordStrength(strengthResult);
+
+  return (
+    <>
+      <Box position="relative">
+        <Input
+          _focus={{ border: '2px solid #5546FF' }}
+          autoFocus
+          border="2px solid"
+          data-testid={OnboardingSelectors.NewPasswordInput}
+          height="64px"
+          key="password-input"
+          placeholder="Set a password"
+          type={showPassword ? 'text' : 'password'}
+          {...field}
+        />
+        <Button
```

Comment:
Remember that `button` defaults to `type="submit"`. As it's a form, this defaults to submitting it. We need a `type="button"` here.